src/components/tl/cuda/allreduce/allreduce_nvls.c

```diff
@@ -84,98 +76,20 @@ void ucc_tl_cuda_allreduce_nvls_progress(ucc_coll_task_t *coll_task)
             task->super.status = UCC_ERR_NO_MEMORY; // TODO: better error code?
             return;
         }
-        cuda_status = cudaEventRecord(
-            ((ucc_ec_cuda_event_t *)task->allreduce_nvls.evtCompletion)->event,
-            stream);
-        if (cuda_status != cudaSuccess) {
-            ucc_error("cudaEventRecord failed: %s",
-                      cudaGetErrorString(cuda_status));
-            task->super.status = UCC_ERR_NO_RESOURCE;
-            return;
-        }
-        task->allreduce_nvls.stage = STAGE_COPY_WAIT;
-        // fallthrough
-    case STAGE_COPY_WAIT:
-        cuda_status = cudaEventQuery(evt);
-        if (cuda_status == cudaErrorNotReady) {
-            task->super.status = UCC_INPROGRESS;
-            return;
-        }
-        if (cuda_status != cudaSuccess) {
-            ucc_error("cudaEventQuery failed %s",
-                      cudaGetErrorString(cuda_status));
-            task->super.status = UCC_ERR_NO_RESOURCE;
-            return;
-        }
-        task->allreduce_nvls.stage = STAGE_COPY_BAR_START;
-        // fallthrough
-    case STAGE_COPY_BAR_START:
-        status = ucc_tl_cuda_shm_barrier_start(trank, task->bar);
-        if (status != UCC_OK) {
-            ucc_error("allreduce barrier start failed");
-            task->super.status = status;
-            return;
-        }
-        task->allreduce_nvls.stage = STAGE_COPY_BAR_TEST;
-        // fallthrough
-    case STAGE_COPY_BAR_TEST:
-        status = ucc_tl_cuda_shm_barrier_test(trank, task->bar);
-        if (status != UCC_OK) {
-            task->super.status = status;
-            return;
-        }
-        task->allreduce_nvls.stage = STAGE_KERNEL_START;
-        // fallthrough
-    case STAGE_KERNEL_START:
+
         status = post_allreduce_kernel(stream, sm_count, threads, mc_va,
                                        task->allreduce_nvls.buf_size_bytes,
-                                       trank, UCC_TL_TEAM_SIZE(team), dt);
+                                       (CUdeviceptr)TASK_NVLS_CONTROL_MC(task),
+                                       (CUdeviceptr)TASK_NVLS_CONTROL_UC(task),
+                                       sm_count * UCC_TL_TEAM_SIZE(team),
```

Comment:
i think this arg is not needed, we already pass sm_count and team_size as separate args